constructor/winexe.py

```diff
@@ -217,27 +210,7 @@ def uninstall_menus_commands(info):
     return [line.strip() for line in lines]
 
 
-def signtool_command(info):
-    "Generates a signtool command to be used in the NSIS template"
-    pfx_certificate = info.get("signing_certificate")
-    if pfx_certificate:
-        signtool = os.environ.get("CONSTRUCTOR_SIGNTOOL_PATH", "signtool")
-        timestamp_server = os.environ.get(
-            "CONSTRUCTOR_SIGNTOOL_TIMESTAMP_SERVER_URL",
-            "http://timestamp.sectigo.com"
-        )
-        command = (
-            f'{str_esc(signtool)} sign /f {str_esc(pfx_certificate)} '
-            f'/tr {str_esc(timestamp_server)} /td sha256 /fd sha256'
-        )
-        if "CONSTRUCTOR_PFX_CERTIFICATE_PASSWORD" in os.environ:
-            # signtool can get the password from the env var on its own
-            command += ' /p "%CONSTRUCTOR_PFX_CERTIFICATE_PASSWORD%"'
-        return command
-    return ""
-
-
-def make_nsi(info, dir_path, extra_files=None, temp_extra_files=None):
+def make_nsi(info, dir_path, extra_files=None, temp_extra_files=None, signing_tool=None):
```

Comment:
Can we add some typing? There are a few things called `signtool` or similar now and it can get confusing. I think this expects an instance of the new class in this PR?